### PR TITLE
ola 0.10.8

### PR DESCRIPTION
**Pull request by peternewman**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting? - I don't have an OS X test machine available anymore
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting? - I don't have an OS X test machine available anymore
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)? - I don't have an OS X test machine available anymore

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
@peternewman, let me know this patch (cdb8f7d) can be removed? it caused the build issue.

**Comment by peternewman** [comment]
> @peternewman, let me know this patch (cdb8f7d) can be removed? it caused the build issue.

Yes it can. This isn't 0.11, but I removed the source of that patch in Macports land!

Should I be deleting the patch file from https://github.com/Homebrew/formula-patches/4dcd2679/ola/libmicrohttpd.diff too?

**Comment by chenrui333** [comment]
```
checking protoc version... 3.6.1
checking google/protobuf/compiler/command_line_interface.h usability... no
checking google/protobuf/compiler/command_line_interface.h presence... yes
configure: WARNING: google/protobuf/compiler/command_line_interface.h: present but cannot be compiled
configure: WARNING: google/protobuf/compiler/command_line_interface.h:     check for missing prerequisite headers?
configure: WARNING: google/protobuf/compiler/command_line_interface.h: see the Autoconf documentation
configure: WARNING: google/protobuf/compiler/command_line_interface.h:     section "Present But Cannot Be Compiled"
configure: WARNING: google/protobuf/compiler/command_line_interface.h: proceeding with the compiler's result
configure: WARNING:     ## --------------------------------------------- ##
configure: WARNING:     ## Report this to open-lighting@googlegroups.com ##
configure: WARNING:     ## --------------------------------------------- ##
checking for google/protobuf/compiler/command_line_interface.h... no
configure: error: Cannot find the protoc header files
```

**Comment by chenrui333** [comment]
> > @peternewman, let me know this patch (cdb8f7d) can be removed? it caused the build issue.
> 
> Yes it can. This isn't 0.11, but I removed the source of that patch in Macports land!
> 
> Should I be deleting the patch file from https://github.com/Homebrew/formula-patches/4dcd2679/ola/libmicrohttpd.diff too?

if we dont use it, let's remove it.

**Comment by peternewman** [comment]
> ```
> checking protoc version... 3.6.1
> checking google/protobuf/compiler/command_line_interface.h usability... no
> checking google/protobuf/compiler/command_line_interface.h presence... yes
> configure: WARNING: google/protobuf/compiler/command_line_interface.h: present but cannot be compiled
> configure: WARNING: google/protobuf/compiler/command_line_interface.h:     check for missing prerequisite headers?
> configure: WARNING: google/protobuf/compiler/command_line_interface.h: see the Autoconf documentation
> configure: WARNING: google/protobuf/compiler/command_line_interface.h:     section "Present But Cannot Be Compiled"
> configure: WARNING: google/protobuf/compiler/command_line_interface.h: proceeding with the compiler's result
> configure: WARNING:     ## --------------------------------------------- ##
> configure: WARNING:     ## Report this to open-lighting@googlegroups.com ##
> configure: WARNING:     ## --------------------------------------------- ##
> checking for google/protobuf/compiler/command_line_interface.h... no
> configure: error: Cannot find the protoc header files
> ```

Odd, that used to work fine, although I'm not sure we've tried on clang 12 before.

**Comment by peternewman** [comment]
Indeed this build was clean 8 days ago on clang 9:
https://travis-ci.org/github/OpenLightingProject/ola/jobs/743671666

**Comment by peternewman** [comment]
I got there in the end @chenrui333 , although the test is failing on 11.0 for some reason, despite compiling fine. I guess `olad -l 4` is failing to auto-launch on that OS, but I can't see why from the logging unfortunately.

File: Formula/ola.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
   homepage "https://www.openlighting.org/ola/"
-  url "https://github.com/OpenLightingProject/ola/releases/download/0.10.7/ola-0.10.7.tar.gz"
-  sha256 "8a65242d95e0622a3553df498e0db323a13e99eeb1accc63a8a2ca8913ab31a0"
+  url "https://github.com/OpenLightingProject/ola/releases/download/0.10.8/ola-0.10.8.tar.gz"
+  sha256 "102aa3114562a2a71dbf7f77d2a0fb9fc47acc35d6248a70b6e831365ca71b13"
   license "GPL-2.0"
-  revision 5
   head "https://github.com/OpenLightingProject/ola.git"
 
   bottle do
```
Comment (peternewman) [inline Formula/ola.rb:9]: Do the bottle checksums need removing/updating, or does that happen automagically?

Comment (chenrui333) [reply Formula/ola.rb:9]: yeah, this got updated automatically, no need to worry about that.

**Comment by chenrui333** [comment]
I have a 11.0 machine, let me build and run the test manually.

**Comment by chenrui333** [comment]
here is the error for the test failure:

```
==> Testing ola
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201123-79702-qgm0w2.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb --debug --verbose
/usr/local/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
No receipt for 'com.apple.pkg.CLTools_Base' found at '/'.
==> /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info
common/network/TCPSocket.cpp:158: connect(127.0.0.1:9010): Connection refused
common/network/TCPSocket.cpp:158: connect(127.0.0.1:9010): Connection refused
examples/ola-client.cpp:930: Setup failed
```

but when I run locally, it seems fine:

```
$ /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info
   Id	Plugin Name
--------------------------------------
    1	Dummy
    2	ArtNet
    3	ShowNet
    4	ESP Net
    5	Serial USB
    6	Enttec Open DMX
    7	SandNet
    8	StageProfi
    9	Pathport
   11	E1.31 (sACN)
   12	USB
   14	OSC
   16	KiNET
   17	KarateLight
   18	Milford Instruments
   19	Renard
   21	Open Pixel Control
   22	GPIO
--------------------------------------
```

**Comment by chenrui333** [comment]
When I changed the test to use `ola_artnet`

the error becomes as:

```
==> /usr/local/Cellar/ola/0.10.8/bin/ola_artnet
Usage: /usr/local/Cellar/ola/0.10.8/bin/ola_artnet -d <dev_id> -n <name> -l <long_name> -s <subnet>

Configure ArtNet devices managed by OLA.

  -d, --device <int32_t>
    Id of the device to control.
  -h, --help
    Display the help message
  -l, --log-level <int8_t>
    Set the logging level 0 .. 4.
  -n, --name <string>
    Set the name of the ArtNet device.
  -s, --subnet <int32_t>
    Set the subnet parameter of the ArtNet device.
  -u, --universe <uint32_t>
    List the IPs of ArtNet devices for this universe.
  -v, --version
    Display version information
  --long-name <string>
    Set the long name of the ArtNet device.
  --net <int32_t>
    Set the net parameter of the ArtNet device.
  --scheduler-policy <string>
    The thread scheduling policy, one of {fifo, rr}.
  --scheduler-priority <uint16_t>
    The thread priority, only used if --scheduler-policy is set.
  --syslog
    Send to syslog rather than stderr.
  --use-kqueue
    Use kqueue() rather than select()
/usr/local/Homebrew/Library/Homebrew/shims/scm/git --version
/usr/bin/curl --version
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
sysctl: unknown oid 'hw.optional.arm64'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:18:in `raise'
BuildError: Failed executing: /usr/local/Cellar/ola/0.10.8/bin/ola_artnet
```

**Comment by chenrui333** [comment]
this would pass the test in my local
```diff
-    system bin/"ola_plugin_info"
+    system bin/"ola_dev_info"
```

**Comment by chenrui333** [comment]
@peternewman build is green now, not sure if you still want to put more updates (i do think there is some underlying issue, but it does not impact the actual usage though)

**Comment by peternewman** [comment]
Thanks @chenrui333 . That's strange! Actually because you haven't given it enough options, ola_dev_info is just showing it's help locally, ola_plugin_info is trying to auto-start olad and then failing somehow.

If you change the test to this, does it give anything else more useful:
`/usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info -l 4`

Also has olad kept running in the background after one of the test runs or manual runs, which could possibly explain the change in behaviour?

They are part of the same codebase ola_plugin_info and ola_dev_info:
https://github.com/OpenLightingProject/ola/blob/e9fcf75fec8a96d2bb91b0f8b41c66765b54aa84/examples/Makefile.mk#L118-L128

**Comment by SMillerDev** [comment]
Seeing those commands at the bottom, does it depend on Java without defining that dependency? Because Big Sur doesn't have a Java version installed.

**Comment by peternewman** [comment]
> Seeing those commands at the bottom, does it depend on Java without defining that dependency? Because Big Sur doesn't have a Java version installed.

We've got a Java plugin @SMillerDev (like the Python and C++ ones) but it's not enabled in this build.

I'm not quite sure which commands you're referring to where, there's no Java in that Makefile, just in the java dir. Edit, just seen it; that and curl in particular aren't OLA I don't think.

@chenrui333 did you get a chance to look at my queries?

**Comment by chenrui333** [comment]
> @chenrui333 did you get a chance to look at my queries?

I thought I did dump the local run in here (looks like I did not), let me reinstall and run the cmd really quick. :)

**Comment by chenrui333** [comment]
looks like we got rid of the `protobuf@3.1`

**Comment by chenrui333** [comment]
rebased to the master and building in my local now.

**Comment by chenrui333** [comment]
@peternewman here is the test output.
```
==> Testing ola
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201219-31984-1qykh76.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb --debug --verbose
/usr/local/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb
==> /usr/local/Cellar/ola/0.10.7_5/bin/ola_plugin_info
common/network/TCPSocket.cpp:158: connect(127.0.0.1:9010): Connection refused
   Id	Plugin Name
--------------------------------------
    1	Dummy
    2	ArtNet
    3	ShowNet
    4	ESP Net
    5	Serial USB
    6	Enttec Open DMX
    7	SandNet
    8	StageProfi
    9	Pathport
   11	E1.31 (sACN)
   12	USB
   14	OSC
   16	KiNET
   17	KarateLight
   18	Milford Instruments
   19	Renard
   21	Open Pixel Control
   22	GPIO
--------------------------------------
NOTE:  Most system logs have moved to a new logging system.  See log(1) for more information.

$ /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info -l 4
-bash: /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info: No such file or directory
```

**Comment by chenrui333** [comment]
take that back, looks like my version did not get updated.

**Comment by chenrui333** [comment]
here is my test result:


<details>
<summary>test result</summary>

```
==> Testing ola
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201219-68542-1vv0u64.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb --debug --verbose
/usr/local/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ola.rb
==> /usr/local/Cellar/ola/0.10.8/bin/ola_dev_info
Device 1: Dummy Device
  port 0, OUT Dummy Port, RDM supported
Device 2: ArtNet [10.0.0.239]
  port 0, IN, priority 100, RDM supported
  port 1, IN, priority 100, RDM supported
  port 2, IN, priority 100, RDM supported
  port 3, IN, priority 100, RDM supported
  port 0, OUT, RDM supported
  port 1, OUT, RDM supported
  port 2, OUT, RDM supported
  port 3, OUT, RDM supported
Device 3: ShowNet [10.0.0.239]
  port 0, IN ShowNet 1-512, priority 100
  port 1, IN ShowNet 513-1024, priority 100
  port 2, IN ShowNet 1025-1536, priority 100
  port 3, IN ShowNet 1537-2048, priority 100
  port 4, IN ShowNet 2049-2560, priority 100
  port 5, IN ShowNet 2561-3072, priority 100
  port 6, IN ShowNet 3073-3584, priority 100
  port 7, IN ShowNet 3585-4096, priority 100
  port 0, OUT ShowNet 1-512
  port 1, OUT ShowNet 513-1024
  port 2, OUT ShowNet 1025-1536
  port 3, OUT ShowNet 1537-2048
  port 4, OUT ShowNet 2049-2560
  port 5, OUT ShowNet 2561-3072
  port 6, OUT ShowNet 3073-3584
  port 7, OUT ShowNet 3585-4096
Device 4: ESP Net [10.0.0.239]
  port 0, IN, priority 100
  port 1, IN, priority 100
  port 2, IN, priority 100
  port 3, IN, priority 100
  port 4, IN, priority 100
  port 0, OUT
  port 1, OUT
  port 2, OUT
  port 3, OUT
  port 4, OUT
Device 5: SandNet [10.0.0.239]
  port 0, IN, priority 100
  port 1, IN, priority 100
  port 2, IN, priority 100
  port 3, IN, priority 100
  port 4, IN, priority 100
  port 5, IN, priority 100
  port 6, IN, priority 100
  port 7, IN, priority 100
  port 0, OUT
  port 1, OUT
Device 6: Pathport [10.0.0.239]
  port 0, IN, priority 100
  port 1, IN, priority 100
  port 2, IN, priority 100
  port 3, IN, priority 100
  port 4, IN, priority 100
  port 5, IN, priority 100
  port 6, IN, priority 100
  port 7, IN, priority 100
  port 0, OUT
  port 1, OUT
  port 2, OUT
  port 3, OUT
  port 4, OUT
  port 5, OUT
  port 6, OUT
  port 7, OUT
Device 7: E1.31 (DMX over ACN) [10.0.0.239]
  port 0, IN, priority inherited
  port 1, IN, priority inherited
  port 2, IN, priority inherited
  port 3, IN, priority inherited
  port 4, IN, priority inherited
  port 0, OUT, priority inherited
  port 1, OUT, priority inherited
  port 2, OUT, priority inherited
  port 3, OUT, priority inherited
  port 4, OUT, priority inherited
Device 8: OSC Device
  port 0, IN /dmx/universe/%d, priority 100
  port 1, IN /dmx/universe/%d, priority 100
  port 2, IN /dmx/universe/%d, priority 100
  port 3, IN /dmx/universe/%d, priority 100
  port 4, IN /dmx/universe/%d, priority 100
Device 9: KiNet Device
NOTE:  Most system logs have moved to a new logging system.  See log(1) for more information.


$ /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info -l 4
   Id	Plugin Name
--------------------------------------
    1	Dummy
    2	ArtNet
    3	ShowNet
    4	ESP Net
    5	Serial USB
    6	Enttec Open DMX
    7	SandNet
    8	StageProfi
    9	Pathport
   11	E1.31 (sACN)
   12	USB
   14	OSC
   16	KiNET
   17	KarateLight
   18	Milford Instruments
   19	Renard
   21	Open Pixel Control
   22	GPIO
--------------------------------------
```

</details>

**Comment by peternewman** [comment]
> looks like we got rid of the `protobuf@3.1`

Yeah, you keep dropping protobuf's. I can't see where we were trying to build with that. Technically we won't build on Python 3.9 until 0.10.9 due to a new bug (probably coming fairly soon due to a few bug reports), so we should perhaps drop back to 3.8, but we didn't have test coverage which is probably how we got bumped. Although some of the RDM stuff doesn't run on Python 3 at all, but again no-one seems to have noticed.

> @peternewman here is my test result:

Looks perfect now, I wonder what changed.

**Comment by peternewman** [comment]
I see both the GitHub Actions are failing to auto launch. But at least one passed in #62874 with no real changes either end that I can see.

**Comment by peternewman** [comment]
Can you check you don't have an olad running after your tests pass.

**Comment by chenrui333** [comment]
> Can you check you don't have an olad process running after your tests pass?

Sure, let me do it now.

**Comment by chenrui333** [comment]
yeah, I dont have any ola process running after the test

```
$ ps -ef | grep ola_
  502 44409 79983   0  6:42PM ttys006    0:00.00 grep ola_
```

**Comment by peternewman** [comment]
> yeah, I dont have any ola process running after the test
> 
> ```
> $ ps -ef | grep ola_
>   502 44409 79983   0  6:42PM ttys006    0:00.00 grep ola_
> ```

Sorry, it would be olad, so `ps -ef | grep ola` would catch everything.

**Comment by peternewman** [comment]
Please can someone call off the @BrewTestBot in https://github.com/Homebrew/formula-patches/pull/322 given it's now closed that linked PR despite my attempt to unstale it.

Is there some way to tell it that it's linked to/blocked behind another PR, otherwise the bot seems a bit flawed?

**Comment by chenrui333** [comment]
it is there.
```
$ ps -ef | grep olad
  502 31998     1   0 Sat01PM ??         0:34.09 olad --daemon --syslog
  502 32821 79983   0 10:14PM ttys006    0:00.00 grep olad
```

**Comment by chenrui333** [comment]
yeah, the behavior is indeed weird.

**Comment by peternewman** [comment]
> it is there.
> 
> ```
> $ ps -ef | grep olad
>   502 31998     1   0 Sat01PM ??         0:34.09 olad --daemon --syslog
>   502 32821 79983   0 10:14PM ttys006    0:00.00 grep olad
> ```

Okay, that explains why your tests are working. If you kill it and re-run the tests I assume they might fail now?

> yeah, the behavior is indeed weird.
> 
> BTW, I have just merged the patch.

Thanks, although I think that breaks the current install as I deleted the (to be redundant) libmicrohttpd patch in that PR.

**Comment by peternewman** [comment]
> Thanks, although I think that breaks the current install as I deleted the (to be redundant) libmicrohttpd patch in that PR.

Scrap that, it's fine due to the magic of commit hashes.

**Comment by chenrui333** [comment]
yeah, after killing `olad`, it shows `no such file or directory`:

```
$ /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info -l 4
-bash: /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info: No such file or directory
```

**Comment by peternewman** [comment]
> yeah, after killing `olad`, it shows `no such file or directory`:
> 
> ```
> $ /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info -l 4
> -bash: /usr/local/Cellar/ola/0.10.8/bin/ola_plugin_info: No such file or directory
> ```

What? :confused: I assume the file does still exist?

It should just show the earlier error about not being able to setup if olad isn't running and it can't auto-launch it:
```
common/network/TCPSocket.cpp:158: connect(127.0.0.1:9010): Connection refused
common/network/TCPSocket.cpp:158: connect(127.0.0.1:9010): Connection refused
examples/ola-client.cpp:930: Setup failed
```

**Comment by peternewman** [comment]
Can we unstale this please, as it's blocked on external input not me?

@chenrui333 did you get a chance to look at my other questions?

> I see both the GitHub Actions are failing to auto launch. But at least one passed in #62874 with no real changes either end that I can see.

I'm still generally at a bit of a loss to this, which is what's causing the tests to fail now.

**Comment by carlocab** [comment]
CI is green. What's the holdup here?

**Comment by peternewman** [comment]
:shrug: @carlocab the OLA tests were failing on the bots, but had arbitrarily started failing due to something external to the change as mentioned in 

It seems to be a bit arbitrary:
> I see both the GitHub Actions are failing to auto launch. But at least one passed in #62874 with no real changes either end that I can see.

But nothing has changed in that regard between this release and the previous one, so I don't really know what's going on, but I'm happy to have it merged. The test command also seemed to run fine for @chenrui333 standalone, just not via homebrew.

I see there's now 11-arm, but we don't seem to be building on that/have a bottle. Do we need to do something to enable it, as we run fine on Arm on Pi and various Debian builds?

**Comment by peternewman** [comment]
I wonder if our test failures are linked to the sandbox, similar to this:
https://github.com/Homebrew/homebrew-core/pull/69794

The test command that was failing is one where a CLI tool launches the main daemon as a separate process, would that be blocked by the sandbox?

**Comment by carlocab** [comment]
I guess I'm just confused by the talk of failures when there's not a failure in sight at the moment.

>would that be blocked by the sandbox?

Not necessarily, but I will admit that I don't quite understand this sandboxing thing.

**Comment by peternewman** [comment]
> I guess I'm just confused by the talk of failures when there's not a failure in sight at the moment. All the tests seem to have passed?

Yes, agreed.

> > would that be blocked by the sandbox?
> 
> Not necessarily, but I will admit that I don't quite understand this sandboxing thing.

Okay, thanks.

> > Do we need to do something to enable it, as we run fine on Arm on Pi and various Debian builds?
> 
> We can attempt a build for it later. We could restart CI for it now to attempt one, but I'm wary of doing that if the tests are as temperamental as the conversation above suggests.

Okay, makes sense. I do seem to get random emails saying they've failed sometimes (I think that's from you not MacPorts) but I've never really got to the bottom of what's causing it. I know we see some race conditions in our `make check` tests on Travis, when the Mac builds actually play ball on Travis, but I don't think Homebrew even runs those tests does it? No one ever complains about the actual apps on Mac.

**Review by carlocab — DISMISSED** [review]
Thanks, @peternewman.

**Comment by carlocab** [comment]
Ok, not sure why that failed.
```
==> Downloading bottles for workflow: tests.yml
Error: No matching workflow run found for these criteria!
```

**Comment by chenrui333** [comment]
rebased to latest master.

**Comment by chenrui333** [comment]
> Ok, not sure why that failed.
> 
> ```
> ==> Downloading bottles for workflow: tests.yml
> Error: No matching workflow run found for these criteria!
> ```

I think the artifacts have expired (rebased and re-generate the bottles would do the job)

**Review by carlocab — APPROVED** [review]

**Comment by peternewman** [comment]
Thanks for the merge, although it looks like building the bottles failed due to protobuf@3.6 for some reason:

**Comment by carlocab** [comment]
That might work. If you can test it, please do, then open a PR to switch the dependency to the newer protobuf + a revision bump.

**Comment by peternewman** [comment]
> That might work. If you can test it, please do, then open a PR to switch the dependency to the newer protobuf + a revision bump.

Unfortunately I can't, as per my comment when opening this PR,
> I don't have an OS X test machine available anymore

Which leaves me a bit stuck supporting it on Homebrew. Especially as our Homebrew via Travis as part of our CI has got rather flaky of late so isn't currently running.

I've opened #71036 in the hope the CI will do the job.

**Pull request closed without merging**